src/language.rs

```diff
@@ -30,161 +30,148 @@ pub enum SupportedLanguageName {
-            name: SupportedLanguageName::Cpp,
-            name_for_ignore_select: "cpp",
-        },
-        SupportedLanguage {
-            language: tree_sitter_go::language(),
-            name: SupportedLanguageName::Go,
-            name_for_ignore_select: "go",
-        },
-        SupportedLanguage {
-            language: tree_sitter_java::language(),
-            name: SupportedLanguageName::Java,
-            name_for_ignore_select: "java",
-        },
-        SupportedLanguage {
-            language: tree_sitter_c_sharp::language(),
-            name: SupportedLanguageName::CSharp,
-            name_for_ignore_select: "csharp",
-        },
-        SupportedLanguage {
-            language: tree_sitter_kotlin::language(),
-            name: SupportedLanguageName::Kotlin,
-            name_for_ignore_select: "kotlin",
-        },
-        SupportedLanguage {
-            language: tree_sitter_elisp::language(),
-            name: SupportedLanguageName::Elisp,
-            name_for_ignore_select: "elisp",
-        },
-        SupportedLanguage {
-            language: tree_sitter_elm::language(),
-            name: SupportedLanguageName::Elm,
-            name_for_ignore_select: "elm",
-        },
-        SupportedLanguage {
-            language: tree_sitter_dockerfile::language(),
-            name: SupportedLanguageName::Dockerfile,
-            name_for_ignore_select: "docker",
-        },
-        SupportedLanguage {
-            language: tree_sitter_html::language(),
-            name: SupportedLanguageName::Html,
-            name_for_ignore_select: "html",
-        },
-        SupportedLanguage {
-            language: tree_sitter_query::language(),
-            name: SupportedLanguageName::TreeSitterQuery,
-            name_for_ignore_select: "treesitterquery",
-        },
-        SupportedLanguage {
-            language: tree_sitter_json::language(),
-            name: SupportedLanguageName::Json,
-            name_for_ignore_select: "json",
-        },
-        SupportedLanguage {
-            language: tree_sitter_css::language(),
-            name: SupportedLanguageName::Css,
-            name_for_ignore_select: "css",
-        },
-        SupportedLanguage {
-            language: tree_sitter_lua::language(),
-            name: SupportedLanguageName::Lua,
-            name_for_ignore_select: "lua",
-        },
+pub static ALL_SUPPORTED_LANGUAGES: BySupportedLanguage<SupportedLanguage> = {
+    use SupportedLanguage::*;
+    [
+        Rust,
+        Typescript,
+        Javascript,
+        Swift,
+        ObjectiveC,
+        Toml,
+        Python,
+        Ruby,
+        C,
+        Cpp,
+        Go,
+        Java,
+        CSharp,
+        Kotlin,
+        Elisp,
+        Elm,
+        Dockerfile,
+        Html,
+        TreeSitterQuery,
+        Json,
+        Css,
+        Lua,
+    ]
+};
+
+static SUPPORTED_LANGUAGE_LANGUAGES: Lazy<BySupportedLanguage<Language>> = Lazy::new(|| {
```

Comment:
...but this one needed a `Lazy` wrapper since I guess `tree_sitter::Language`'s aren't `const`?